sequence/src/main/java/com/vitreoussoftware/bioinformatics/sequence/io/reader/embl/EmblStringFileStreamReader.java

```diff
@@ -0,0 +1,260 @@
+	 * @param file the file to run on
+	 */
+	private EmblStringFileStreamReader(FileReader file, int pagingSize)
+	{
+		this.file = file;
+
+		buffer = new char[pagingSize];
+		length = 0;  // we don't have anything in the buffer
+		index = 0;   // index starts at 0
+	}
+
+    /**
+     * Create an input stream for EMBL file format
+     * @param fileName the EMBL file
+     * @return the input stream
+     * @throws FileNotFoundException the specified file was not found
+     */
+    public static StringStreamReader create(String fileName) throws FileNotFoundException {
+        return create(fileName, DEFAULT_BUFFER_SIZE);
+    }
+
+    /**
+     * Create an input stream for EMBL file format
+     * @param file the EMBL file
+     * @return the input stream
+     * @throws FileNotFoundException the specified file was not found
+     */
+    public static StringStreamReader create(FileReader file) throws FileNotFoundException {
+        return create(file, DEFAULT_BUFFER_SIZE);
+    }
+
+    /**
+     * Create an input stream for EMBL file format
+     * @param file the EMBL file
+     * @param pagingSize the size of the buffer for paging data from disk
+     * @return the input stream
+     * @throws FileNotFoundException the specified file was not found
+     */
+    public static StringStreamReader create(FileReader file, int pagingSize) throws FileNotFoundException {
+        return new EmblStringFileStreamReader(file, pagingSize);
+    }
+
+    /**
+     * Create an input stream for EMBL file format
+     * @param fileName the EMBL file
+     * @param pagingSize the size of the buffer for paging data from disk
+     * @return the input stream
+     * @throws FileNotFoundException the specified file was not found
+     */
+    public static StringStreamReader create(String fileName, int pagingSize) throws FileNotFoundException {
+        FileReader file = new FileReader(fileName);
+        return new EmblStringFileStreamReader(file, pagingSize);
+    }
+
+	/**
+     * Reads a record from the file
+	 * @return the record
+	 */
+	@Override
+	public Pair<String,String> next() {
+        String metadata = readMetadata();
+        String data = readSequenceData();
+
+        return Pair.with(metadata, data);
+    }
+
+	@Override
+	protected void finalize() throws Throwable {
+		this.file.close();
+		super.finalize();
+	}
+
+	/**
+     * Does the stream reader still have a record?
+	 * @return boolean indicator
+	 * @throws IOException If the file cannot be accessed it may fail
+	 */
+	@Override
+	public boolean hasNext() {
+        // Dump whitespace
+        dropUntil(x -> !Character.isWhitespace(x));
+
+		// if we know we have data remaining say so
+		if (length > 0 && index < length)
+			return true;
+
+		// if we don't know find out
+		bufferData();
+		
+		return length > 0 && index < length;
+	}
+
+	@Override
+	public void close() throws IOException {
+		this.file.close();
+	}
+
+    private void dropUntil(String flag) {
```

Comment:
Refactored out into BufferedFileStreamReader class to provide help for performing buffered reads from files.